### ISSUE TITLE
Multiple Updates in one transaction!

### ISSUE THREAD

**Issue by nikatlas** [aaronpowell/db.js#123]
Hi there, 

I am using db.js for large data sets and experienced delayed behaviour on Update method. 
Specifically writing 10000 objects serially took me more than 20 seconds. 

Examining the code i found that the add method can take an Array of objects as parameter: 
```javascript
this.add = function (table) {
            if (closed) {
                throw 'Database has been closed';
            }

            var records = [];
            var counter = 0;

            for (var i = 0; i < arguments.length - 1; i++) {
                if (Array.isArray(arguments[i + 1])) {
                    for (var j = 0; j < (arguments[i + 1]).length; j++) {
                        records[counter] = (arguments[i + 1])[j];
                        counter++;
                    }
                } else {
                    records[counter] = arguments[i + 1];
                    counter++;
                }
            }
...
...
````

The update method doesnt have this argument parsing loop. It actually looks like this : 
````javascript
this.update = function (table) {
            if (closed) {
                throw 'Database has been closed';
            }
            var records = [];
            for (var i = 0 ; i < arguments.length - 1 ; i++) {
                records[i] = arguments[i + 1];
            }
....
...
..
````

First of all is there a reason which update cant handle arrays ? maybe i am missing something.  
I actually wanted to write in one transaction and switching the for loop took me from 20+ seconds to 500ms (maybe less)

I can send a pull-merge request if this is ok.

**Comment by aaronpowell** [comment]
It's probably just an oversight. You can either use `apply` on the `update` method and pass an array or send through a PR

**Comment by brettz9** [comment]
Any update on the proposed PR?

One optimization that should help with performance would be to cache the `arguments.length` so as to avoid retrieving from the `arguments` object upon each iteration. One might also cache `arguments[i + 1]` and `arguments[i + 1]).length` as well as Array.isArray into a local variable as well.

FTR, this issue also ties in to issue #95 .

**Comment by brettz9** [comment]
I think this issue can now be closed as the array updating was addressed in the now merged PR #133. (As far as other use cases for handling updates in one transaction, that is already being tracked by #95 .)

**Comment by brettz9** [comment]
Not sure if you missed that this one can also be closed (though it is possible for multiple updates to be run in another way but that is being tracked by #95 ).
